src/nvim/api/ui.c

```diff
@@ -24,21 +23,13 @@ typedef struct {
 static PMap(uint64_t) *connected_uis = NULL;
 
 void remote_ui_init(void)
+    FUNC_ATTR_NOEXPORT
 {
   connected_uis = pmap_new(uint64_t)();
-  // Add handler for "attach_ui"
-  String method = cstr_as_string("ui_attach");
-  MsgpackRpcRequestHandler handler = {.fn = remote_ui_attach, .async = false};
-  msgpack_rpc_add_method_handler(method, handler);
-  method = cstr_as_string("ui_detach");
-  handler.fn = remote_ui_detach;
-  msgpack_rpc_add_method_handler(method, handler);
-  method = cstr_as_string("ui_try_resize");
-  handler.fn = remote_ui_try_resize;
-  msgpack_rpc_add_method_handler(method, handler);
 }
 
 void remote_ui_disconnect(uint64_t channel_id)
+    FUNC_ATTR_NOEXPORT
```

Comment:
There was a reason FUNC_ATTR_ was used, it is understood by our tooling, like `gendeclarations.lua`, so arguably they are not strictly only for the compiler. I could add the special cases to gendeclarations as well but these will be special cases that needs to remembered by anything else wanting to parse neovim code.
An option would be to let `FUNC_` be the generic prefix and reserve `FUNC_ATTR_` for compiler attributes, and use say `FUNC_API_ASYNC` and `FUNC_API_NOEXPORT`.